Mono.Debugging.Soft/SoftDebuggerSession.cs

```diff
@@ -1212,20 +1215,88 @@ void InsertCatchpoint (Catchpoint cp, BreakInfo bi, TypeMirror excType)
 		{
 			ExceptionEventRequest request;
 
-			request = vm.CreateExceptionRequest (excType, true, true);
-			//Commenting Count so we have better control of counting
-			//because VM only allows count equal to some number but we need also
-			//lower, greater, equal or greater...
-			//Plus some day we might want to put filtering before counting...
-			//request.Count = cp.HitCount; // Note: need to set HitCount *before* enabling
+			//If the VM doesn't support the EverythingElse feature, Other Exceptions will not be caught
+			//In this case the user will have to set the exceptions he wants to catch manually
+			if (!vm.Version.AtLeast (2, 54) && cp.EverythingElse)
+				return;
+
+			request = vm.CreateExceptionRequest (excType, true, true, cp.EverythingElse);
+			//Commenting Count so we have better control of counting
+			//because VM only allows count equal to some number but we need also
+			//lower, greater, equal or greater...
+			//Plus some day we might want to put filtering before counting...
+			//request.Count = cp.HitCount; // Note: need to set HitCount *before* enabling
 			if (vm.Version.AtLeast (2, 25))
 				request.IncludeSubclasses = cp.IncludeSubclasses; // Note: need to set IncludeSubclasses *before* enabling
 			request.Enabled = cp.Enabled;
 			bi.Requests.Add (request, excType.Assembly);
 
 			breakpoints[request] = bi;
+		}
+
+		#region ExceptionRequest
+		//VS in Windows doesn't have the concept of a Catchpoint. In the past we did use it anyway, but that leads to all sort of issues and bugs.
+		//The most important difference is that subclasses are never caught, and that Others is a concept that the Catchpoint doesn't support properly.
+		//This region is for Windows, to be able to properly handle exceptions with an exception list. The concept of Other is supported starting with vm version 2.54.
+
+		Dictionary<string, ExceptionEventRequest> exceptionRequests = new Dictionary<string, ExceptionEventRequest>();
+		ExceptionEventRequest otherExceptions;
```

Comment:
This needs to be reset back to `null` when the session ends or it will be out-of-sync if/when the session gets reused.

Reply:
Correcting